bokeh/application/application.py

```diff
@@ -100,9 +101,23 @@ class Application(object):
     # will function without bringing in tornado.
     _is_a_bokeh_application_class = True
 
-    def __init__(self, *handlers):
+    def __init__(self, *handlers, metadata=None):
```

Comment:
Sure that's also fine, we do something similar in many places. The nicest version would check for any additional unexpected kwargs and raise an exception.